### PR TITLE
finished direct message feature up to step 4

### PR DESCRIPTION
**Pull request by andreayang310**

### CODE REVIEW

**Comment by ctrokid** [comment]
This looks great, Andrea. A few things before I grant you approval.

Do you see that the "all checks have failed" warning is apparent? I think you should read through the `README.md` in the project, and hook up the checkstyle linter as your github pre-commit check (should be detailed in the readme - let me know if you need help).

After that, whenever you run `git commit -m "my message"`, the linter will check your style before allowing you to commit.

I will be happy to grant approval after the style issues are fixed. That way, I won't duplicate any comments that the linter will already warn you about.

**Comment by andreayang310** [comment]
@ctrokid Hi Collin. I completely overlooked linter errors yesterday, my bad. I've added a commit after fixing them. (And I've set up the linter and it works now)

File: src/main/java/com/google/codeu/data/Message.java
Comment (ctrokid) [inline src/main/java/com/google/codeu/data/Message.java:31]: Thanks a lot for updating the comments/documentation. This is very valuable in industry.

File: src/test/java/com/google/codeu/data/MessageTest.java
Comment (ctrokid) [inline src/test/java/com/google/codeu/data/MessageTest.java:19]: +1 for updating the tests, too!

File: src/main/java/com/google/codeu/servlets/MessageServlet.java
Comment (ctrokid) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java:86]: What is the exact motivation behind this change?

**Review by ctrokid — CHANGES_REQUESTED** [review]
This looks great. I will approve after you answer my one silly question (I'm just trying to give you the hang of what a code review looks/feels like).

File: src/main/java/com/google/codeu/data/Message.java
Comment (ctrokid) [reply src/main/java/com/google/codeu/data/Message.java:31]: For these where I don't suggest anything in specific, you can either leave it open (not recommended), or you can leave a comment like "ack" or acknowledged or "ok" to state that you've seen the comment (then click "resolve conversation").

Comment (andreayang310) [reply src/main/java/com/google/codeu/data/Message.java:31]: ok

File: src/main/java/com/google/codeu/servlets/MessageServlet.java
Comment (andreayang310) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java:86]: From my understanding it's so that the user returns to the page of the recipient, which is where they started from, and not their own page?

Comment (ctrokid) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java:86]: Thanks for the explanation, that makes sense to me.

**Review by ctrokid — APPROVED** [review]

**Pull request merged**
